### DISCUSSION TITLE
Installing genealogy on clean install with Windows + WSL + Ubuntu + MySQL [SIMPLE GUIDE]

### DISCUSSION THREAD

**Discussion by adamgas** [MGeurts/genealogy#28] [Show and tell]
Prerequisites:

- newest Windows 11
- WSL installed
- WSL default set to WSL2

---

Open PowerShell

Install Ubuntu 24
```
wsl --install --distribution Ubuntu-24.04
```

Run Ubuntu 24. Everything else is done in Ubuntu:

---

Install php:
```
sudo apt install php libapache2-mod-php php-mbstring php-xmlrpc php-soap php-gd php-xml php-cli php-zip php-bcmath php-tokenizer php-json php-pear php-mysql
```
Probably there are too many things up here, but now I cannot check which ones are necessary.

Install Composer:
```
sudo apt install composer
```

Install npm:
```
sudo apt install npm
```

Install MySQL:
```
sudo apt install mysql-server
```

---
In MySQL 
```
sudo mysql
```
do two things:

1. Create database
2. Create database user

Create database with:
```
create database genealogy;
```

Create database user, according to https://www.digitalocean.com/community/tutorials/how-to-create-a-new-user-and-grant-permissions-in-mysql

Grant ALL PRIVILEGES (without GRANT OPTION) to 'gen'@'localhost' user at any database schema and any table or other database object. Maybe it's too much - **consider with care.**

```
grant ALL PRIVILEGES ON *.* TO 'gen'@'localhost';
```

They say it helps:
```
FLUSH PRIVILEGES;
```

---
Because of this error:
https://stackoverflow.com/questions/35240414/laravel-5-pdoexception-could-not-find-driver

Go to php.ini file
```
sudo nano /etc/php/8.3/apache2/php.ini
```

Find:
```
;extension=pdo_mysql.so
```

Uncomment this line, so it is:
```
extension=pdo_mysql.so
```

Then restart Apache2 server:
```
sudo service apache2 restart
```

---
Install `genealogy` according to installation rules in README

create a new project folder, cd into the folder
```
git clone https://github.com/MGeurts/genealogy.git .
```
```
cp .env.example .env
```

make the needed changes regarding name, url, database connection & mail server

>In my case i edited only these lines:
```
DB_USERNAME=gen
DB_PASSWORD=***password set when creating gen user***
```

```
composer install
```
```
php artisan key:generate
```
```
php artisan storage:link
```
```
php artisan migrate:fresh --seed
```
```
npm install & npm run build
```

Run genealogy app with:
`php artisan serve` or `npm run dev`

---

Open your browser. Type in:
```
http://127.0.0.1:8000
```

The application should be running. It runs locally on the computer, accessible from Windows environment.

**Comment by MGeurts** [comment]
Thank you for contributing this solution.

Me myself, I prefer the simplicity of using [LARAGON](https://laragon.org/) as a Dev Stack.

When configuring PHP, make sure you have the ZIP extension installed and enabled.

![LARAGON - PHP Extensions](https://github.com/user-attachments/assets/0770bcac-6d3e-492a-a818-5d93217cbfc8)

**Reply by chinmaypurav** [reply]
@MGeurts I would recommend we add these dependencies inside composer.json
